File: RallyGO/App/rallygo/rallygo/app/Models/Ranking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Ranking extends Model
{
    // Nombre de la tabla en la base de datos
    protected $table = 'ranking';

    // La tabla no tiene created_at ni updated_at
    public $timestamps = false;

    // Campos que se pueden llenar masivamente
    protected $fillable = [
        'posicion',
        'piloto',
        'puntos',
        'equipo'
    ];
}

File: RallyGO/App/rallygo/lib/models/ranking.dart
// lib/models/ranking.dart
class Ranking {
  final int posicion;
  final String piloto;
  final int puntos;
  final String equipo;

  Ranking({
    required this.posicion,
    required this.piloto,
    required this.puntos,
    required this.equipo,
  });

  factory Ranking.fromJson(Map<String, dynamic> json) {
    return Ranking(
      posicion: json['posicion'],
      piloto: json['piloto'],
      puntos: json['puntos'],
      equipo: json['equipo'],
    );
  }
}
